### PR TITLE
Azure: Install Azure CLI

### PR DESCRIPTION
**Pull request by chewong**
[`az acr build`](https://docs.microsoft.com/en-us/cli/azure/acr?view=azure-cli-latest#az-acr-build) will allow us to build and push Windows containers in a Linux environment (prow) ([example](https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/334#issuecomment-605082289)). Installing Azure CLI will be the first step in enabling CSI driver e2e test on Windows clusters. 

/cc @aramase @andyzhangx @ZeroMagic 
/assign @feiskyer

### CODE REVIEW

File: config/jobs/kubernetes-sigs/cloud-provider-azure/cloud-provider-azure-config.yaml [hunk 1]
```diff
@@ -263,7 +263,7 @@ periodics:
       - --aksengine-deploy-custom-k8s
       - --aksengine-location=westus2
       - --aksengine-public-key=$(AZURE_SSH_PUBLIC_KEY_FILE)
-      - --aksengine-template-url=https://raw.githubusercontent.com/kubernetes-sigs/cloud-provider-azure/master/tests/k8s-azure/manifest/autoscaler-virtual-kubelet.json
+      - --aksengine-template-url=https://raw.githubusercontent.com/kubernetes-sigs/cloud-provider-azure/master/tests/k8s-azure/manifest/linux-autoscaler.json
```
Comment (chewong) [inline config/jobs/kubernetes-sigs/cloud-provider-azure/cloud-provider-azure-config.yaml]: Creating 1.18 cluster with https://raw.githubusercontent.com/kubernetes-sigs/cloud-provider-azure/master/tests/k8s-azure/manifest/autoscaler-virtual-kubelet.json and aks-engine 0.48.0 will have the following ARM error:
```
Custom data in OSProfile must be in Base64 encoding and with a maximum length of 87380 characters
```
Temporarily changing it to https://raw.githubusercontent.com/kubernetes-sigs/cloud-provider-azure/master/tests/k8s-azure/manifest/linux-autoscaler.json while investigating the error.

File: kubetest/azure.go [hunk 2]
```diff
@@ -424,9 +424,30 @@ func newAKSEngine() (*Cluster, error) {
...
 
+	if err := c.installAzureCLI(); err != nil {
+		return nil, err
+	}
+
 	return &c, nil
 }
 
+func (c *Cluster) installAzureCLI() error {
+	if err := control.FinishRunning(exec.Command("pip", "install", "azure-cli")); err != nil {
```
Comment (feiskyer) [inline kubetest/azure.go]: could we check whether "az" exist before install? we shouldn't override existing Azure CLI if it has already been installed.

File: config/jobs/kubernetes-sigs/cloud-provider-azure/cloud-provider-azure-config.yaml (hunk 1 above)
Comment (feiskyer) [reply config/jobs/kubernetes-sigs/cloud-provider-azure/cloud-provider-azure-config.yaml]: @ritazh anything changed from aks-engine?

Comment (ritazh) [reply config/jobs/kubernetes-sigs/cloud-provider-azure/cloud-provider-azure-config.yaml]: Just more custom data installed by default from all the addons. This happens when the custom data has reached cloud-init max size.

**Pull request merged**